src/backend/revoker/subScan.ts

```diff
@@ -12,69 +12,141 @@ const SUBSCAN_MAX_ROWS = 100;
 const QUERY_INTERVAL_MS = 1000;
 const BLOCK_RANGE_SIZE = 100_000;
 
-const eventsApi = `https://${subscan.network}.api.subscan.io/api/scan/events`;
+const eventsListURL = `https://${subscan.network}.api.subscan.io/api/v2/scan/events`;
+const eventsParamsURL = `https://${subscan.network}.api.subscan.io/api/scan/event/params`;
 const headers = {
   'X-API-Key': subscan.secret,
 };
 
-export interface EventsResponseJson {
+/**
+ * Structure of SubScan responses from `/api/v2/scan/events`.
+ */
+export interface EventsListJSON {
+  code?: number;
   data: {
     count: number;
-    events: Array<{
-      params: string;
-      block_num: number;
-      block_timestamp: number;
-      extrinsic_hash: HexString;
-    }> | null;
+    events:
+      | {
+          block_timestamp: number; // UNIX-time in seconds
+          event_id: string;
+          event_index: string;
+          extrinsic_hash: string;
+          extrinsic_index: string;
+          finalized: true;
+          id: number;
+          module_id: string;
+          phase: number;
+        }[]
+      | null;
   };
+  generated_at?: number;
+  message?: string;
+}
+
+/**
+ * Structure of SubScan responses from `/api/scan/event/params`.
+ */
+export interface EventsParamsJSON {
+  code: number;
+  data: {
+    event_index: string;
+    params: {
+      name?: string;
+      type?: string;
+      type_name: string;
+      value: unknown;
+    }[];
+  }[];
+  generated_at: number;
+  message: string;
 }
 
 export async function getEvents({
   fromBlock,
   row = SUBSCAN_MAX_ROWS,
   ...parameters
 }: {
-  module: string;
-  call: string;
+  module: string; // Pallet name
+  event_id: string; // Event emitted
   fromBlock: number;
   page: number;
   row?: number;
 }) {
-  const json = {
+  const payloadForEventsListRequest = {
     ...parameters,
     block_range: `${fromBlock}-${fromBlock + BLOCK_RANGE_SIZE}`,
+    order: 'asc',
     row,
     finalized: true,
   };
 
+  logger.debug('payloadForEventsListRequest: ' + payloadForEventsListRequest);
+
   const {
     data: { count, events },
-  } = await got.post(eventsApi, { headers, json }).json<EventsResponseJson>();
+  } = await got
+    .post(eventsListURL, { headers, json: payloadForEventsListRequest })
+    .json<EventsListJSON>();
 
   if (!events) {
     return { count };
   }
 
-  events.reverse();
+  const eventIndices = events.map((event) => event.event_index);
+
+  const payloadForEventsParamsRequest = { event_index: eventIndices };
+  logger.debug(
+    'payloadForEventsParamsRequest: ' + payloadForEventsParamsRequest,
+  );
+
+  const { data: eventsDetails } = await got
+    .post(eventsParamsURL, { headers, json: payloadForEventsParamsRequest })
```

Comment:
Since we asked for event params, why do we store it as event details?

Reply:
Did it on commit: [bdcc6c9](https://github.com/BTE-Trusted-Entity/socialkyc.io/pull/1957/commits/bdcc6c9d82b3e20d6ed1c7110a4fefa0c9138f76)